flink-core/src/main/java/org/apache/flink/configuration/CheckpointingOptions.java

```diff
@@ -109,6 +109,20 @@ public class CheckpointingOptions {
                     .defaultValue(1)
                     .withDescription("The maximum number of completed checkpoints to retain.");
 
+    /**
+     * Option whether to clean individual checkpoint's operatorstates in parallel. If enabled,
+     * operator states are discarded in parallel using the ExecutorService passed to the cleaner.
+     * This speeds up checkpoints cleaning, but adds load to the IO.
+     */
+    @Documentation.Section(Documentation.Sections.COMMON_STATE_BACKENDS)
+    public static final ConfigOption<Boolean> CLEANER_PARALLEL_MODE =
+            ConfigOptions.key("state.checkpoint.cleaner.parallel-mode")
+                    .booleanType()
+                    .defaultValue(false)
```

Comment:
Why not `defalutValue(true)`?

Reply:
so instead of a new configuration, I updated the change with discardAsync that will just discard the state objects in parallel. WDYT?

Reply:
Ahh, sorry for missleading. I meant to keep the config option, but make the default value true, as this seems to be universally positive change, unless there is a bug/some unforeseen complication.

Reply:
@pnowojski can you advise if it is better to have an option to keep the original behavior? thank you!

Reply:
I would keep more or else the old code path, for example if there is a bug or some unexpected behaviour, so that users could disable this feature. Otherwise I think it should be enabled by default and if nobody complains, in next or the following release we could drop the flag to force sequential clean up (so we would only have parallel clean up).

Reply:
add back the config and set default value to true. @pnowojski can you review again? thank you for your patient reviews!